src/main/java/com/sun/caresyncsystem/service/impl/ProfileServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.sun.caresyncsystem.service.impl;
+
+import com.sun.caresyncsystem.dto.request.UpdateProfileRequest;
+import com.sun.caresyncsystem.dto.response.UserProfileResponse;
+import com.sun.caresyncsystem.exception.AppException;
+import com.sun.caresyncsystem.exception.ErrorCode;
+import com.sun.caresyncsystem.mapper.ToDtoMappers;
+import com.sun.caresyncsystem.model.entity.Doctor;
+import com.sun.caresyncsystem.model.entity.Patient;
+import com.sun.caresyncsystem.model.entity.User;
+import com.sun.caresyncsystem.repository.DoctorRepository;
+import com.sun.caresyncsystem.repository.PatientRepository;
+import com.sun.caresyncsystem.repository.UserRepository;
+import com.sun.caresyncsystem.service.AuthenticationService;
+import com.sun.caresyncsystem.service.ProfileService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class ProfileServiceImpl implements ProfileService {
+
+    private final UserRepository userRepository;
+    private final DoctorRepository doctorRepository;
+    private final PatientRepository patientRepository;
+    private final AuthenticationService authenticationService;
+
+    @Override
+    public UserProfileResponse getCurrentUserProfile() {
+        User currentUser = getAuthenticatedUser();
+
+        Doctor doctor = doctorRepository.findByUserId(currentUser.getId()).orElse(null);
+        Patient patient = patientRepository.findByUserId(currentUser.getId()).orElse(null);
+
+        return ToDtoMappers.toUserProfileResponse(currentUser, doctor, patient);
+    }
+
+    @Override
+    @Transactional
+    public UserProfileResponse updateProfile(UpdateProfileRequest request) {
+        User user = getAuthenticatedUser();
+
+        updateUserBasicInfo(user, request);
+        userRepository.save(user);
+
+        Doctor doctor = null;
+        Patient patient = null;
+
+        switch (user.getRole()) {
+            case DOCTOR -> doctor = handleDoctorProfileUpdate(user, request);
+            case PATIENT -> patient = handlePatientProfileUpdate(user, request);
+            default -> throw new AppException(ErrorCode.UNAUTHORIZED);
+        }
+
+        return ToDtoMappers.toUserProfileResponse(user, doctor, patient);
+    }
+
+    private User getAuthenticatedUser() {
+        return authenticationService.getCurrentUser();
+    }
```

Comment:
sao không gọi thẳng
```suggestion
User user = authenticationService.getCurrentUser();
    }
```
mà phải tạo thêm fun e nhĩ? hơi thừa qué ta

Reply:
e thấy nhiều chỗ dùng lại nên e tách á a

Reply:
@nampx-2246 thì biết là nhiều chỗ dùng, ý a là :
User user = authenticationService.getCurrentUser(); 
không nhanh hơn so với
 private User getAuthenticatedUser() {
        return authenticationService.getCurrentUser();
    }
+
User currentUser = getAuthenticatedUser();
hay sao e

Reply:
k tách cũng dc nhưng IDE báo warning ngứa mắt a ạ :v